python/perf-kernels/multreduce_matmul_kernel.py

```diff
@@ -0,0 +1,73 @@
+import triton
+import triton.language as tl
+
+
+# Kernel that implements GEMM with explicit multiply-reduce instructions for small block sizes.
+# Based on **tune_gemm** `matmul_kernel` from commit `cf44637` (see `triton-mlir` branch).
+@triton.jit
+def multreduce_matmul_kernel(a_ptr, b_ptr, c_ptr, bias_ptr, M, N, K, stride_am, stride_ak, stride_bk, stride_bn,
+                             stride_cm, stride_cn, stride_bias, BLOCK_SIZE_M: tl.constexpr, BLOCK_SIZE_N: tl.constexpr,
+                             BLOCK_SIZE_K: tl.constexpr, SPLIT_K: tl.constexpr, GROUP_SIZE_M: tl.constexpr,
+                             BIAS: tl.constexpr, EVEN_K: tl.constexpr):
+    pid = tl.program_id(axis=0)
+    pid_z = tl.program_id(1)
+    num_pid_m = tl.cdiv(M, BLOCK_SIZE_M)
+    num_pid_n = tl.cdiv(N, BLOCK_SIZE_N)
+    if GROUP_SIZE_M == 1:
+        pid_m = pid // num_pid_n
+        pid_n = pid % num_pid_n
+    else:
+        num_pid_in_group = GROUP_SIZE_M * num_pid_n
+        group_id = pid // num_pid_in_group
+        first_pid_m = group_id * GROUP_SIZE_M
+        group_size_m = min(num_pid_m - first_pid_m, GROUP_SIZE_M)
+        pid_m = first_pid_m + (pid % group_size_m)
+        pid_n = (pid % num_pid_in_group) // group_size_m
+    if SPLIT_K == 1:
```

Comment:
Can you remove the other GEMM code like splitK and code for large M/N?

Reply:
Of course, I can remove these pieces. Please let me remove them in another development branch and check correctness again. After that, I'm going to update this PR.

Reply:
Now it's done. `SPLIT_K` was removed by 9cbebf09ce9a11b31890dd21a4a1e15babbb28ca and `GROUP_SIZE_M` was removed by dd161e502a3bbc6cabfae9ded4d006bdadd6973c.